src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,89 @@
+
+  useEffect(() => {
+    getUser(selectedModal.userId)
+      .then(setUser)
+      .catch(() => setHasLoadingModalError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
-
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button
-              type="button"
-              className="delete"
-              data-cy="modal-close"
-            />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
-              {' by '}
-
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
-            </p>
+      {isLoding
+        ? (
+          <Loader />
+        )
+        : (
+          <div className="modal-card">
+            {hasLoadingModalError
+              ? (
+                <div className="notification is-danger">
+                  An error occured when loading data!
+                </div>
+              )
+              : (
+                <>
+                  <header className="modal-card-head">
+                    <div
+                      className="modal-card-title has-text-weight-medium"
+                      data-cy="modal-header"
+                    >
+                      {`Todo #${selectedModal.id}`}
+                    </div>
+
+                    <button
+                      type="button"
+                      aria-label="Mute volume"
+                      className="delete"
+                      data-cy="modal-close"
+                      onClick={() => onSelectedModal(null)}
+                    />
+                  </header>
+
+                  <div className="modal-card-body">
+                    <p className="block" data-cy="modal-title">
+                      {selectedModal.title}
+                    </p>
+
+                    <p className="block" data-cy="modal-user">
+                      {selectedModal.completed
+                        ? (
+                          <strong className="has-text-success">Done</strong>
+                        )
+                        : (
+                          <strong className="has-text-danger">Planned</strong>
+                        )}
```

Comment:
make use of classNames + ternary operator

Reply:
agree with you, the classnames looks better, but then all my buttons get this class, not only pushed button.